In [1]:
import pandas



In [2]:
df = pandas.read_csv("cars.csv")



In [3]:
X = df[['Weight', 'Volume']]
y = df['CO2']

In [4]:
from sklearn import linear_model

regr = linear_model.LinearRegression()
regr.fit(X, y)

LinearRegression(copy_X=True, fit_intercept=True, n_jobs=None, normalize=False)

In [5]:
#predict the CO2 emission of a car where the weight is 2300kg, and the volume is 1300cm3:
predictedCO2 = regr.predict([[2300, 1300]])

In [6]:
predictedCO2

array([107.2087328])

In [8]:
print(regr.coef_) # if x is a variable, then 2x is x two times. 
                  # x is the unknown variable, and the number 2 is the coefficient.

[0.00755095 0.00780526]


### Weight: 0.00755095
### Volume: 0.00780526

### These values tell us that if the weight increase by 1kg, the CO2 emission 
### increases by 0.00755095g.

### And if the engine size (Volume) increases by 1 cm3, the CO2 emission increases by 0.00780526 g.